### PR TITLE
Serve JSON update manifests instead of RDF

### LINKED ISSUE #7223: Serve JSON update manifests for clients that support it
Firefox has supported a [JSON add-on update manifest](https://developer.mozilla.org/en-US/Add-ons/Updates) since version 45, but support for the old RDF format can't be dropped until add-ons stop using it. AMO needs to start serving the JSON updates for clients that support it.

The first step is figuring out how to determine which manifest to serve. Is there an `Accept` header that can be used, or is the user agent needed?

### PR DESCRIPTION
**Pull request by diox**
Also remove code to support hotfixes for super old versions of Firefox

Fix #7223

### CODE REVIEW

File: services/update.py [hunk 1]
```diff
@@ -243,47 +190,62 @@ def get_update(self):
 
         return False
 
-    def get_bad_rdf(self):
-        return bad_rdf
-
-    def get_rdf(self):
+    def get_output(self):
```
Comment (eviljeff) [inline services/update.py]: https://github.com/mozilla/addons-server/pull/8175/files#diff-a3aebfacc7f05d955e54b62d722ae70aL305 still references `get_rdf`

Comment (eviljeff) [reply services/update.py]: is an integration test possible to catch that kind of mistake?

File: services/update.py [hunk 2]
```diff
@@ -243,47 +190,62 @@ def get_update(self):
...
         self.cursor.close()
         if self.conn:
             self.conn.close()
-        return rdf
+        return json.dumps(output)
 
-    def get_no_updates_rdf(self):
-        name = base.ADDON_SLUGS_UPDATE[self.data['type']]
-        return no_updates_rdf % ({'guid': self.data['guid'], 'type': name})
+    def get_error_output(self):
```
Comment (eviljeff) [inline services/update.py]: Is this what Firefox expects for an error output?  The MDN doc was vague.  (It's a sensible error output, just wondering)

File: services/update.py (hunk 1 above)
Comment (diox) [reply services/update.py]: doh! I can probably try to call `application()` directly

File: src/olympia/addons/tests/test_update.py [hunk 3]
```diff
@@ -638,147 +641,71 @@ def get_file_url(self):
...
         version.update(releasenotes=None)
 
-        rdf = self.get(self.good_data).get_rdf()
-        assert rdf.find('updateInfoURL') == -1
+        content = self.get_update_instance(self.good_data).get_output()
+        data = json.loads(content)
+        assert 'update_info_url' not in data['addons'][guid]['updates'][0]
 
-    def test_sea_monkey(self):
+    def test_seamonkey(self):
```
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/addons/tests/test_update.py]: It doesn't, but the test was there so...

File: services/update.py (hunk 2 above)
Comment (diox) [reply services/update.py]: I'm not sure what Firefox expects either. I went with what made the most sense, and we'll see if that works... tbh I don't think it matters much - Firefox doesn't stupidly crash if we return nonsense, worst case scenario it logs something to the browser console, which isn't a bad thing in this situation.

File: src/olympia/addons/tests/test_update.py [hunk 4]
```diff
@@ -416,6 +418,32 @@ def check(self, expected):
...
+                ('Last-Modified', 'Fri, 04 May 2018 11:27:00 GMT'),
+                ('Expires', 'Fri, 04 May 2018 12:27:00 GMT'),
+                ('Content-Length', '2')
+            ]
+
+        output = update.application(environ, start_response_inspector)
+        assert self.start_response_call_count == 1
+        # Output is an array with a single string containing the body of the
+        # response - in this case an empty json dict.
+        assert output == ['{}']
```
Comment (eviljeff) [inline src/olympia/addons/tests/test_update.py]: An end to end that at least produced a no_updates_output would probably be better - the error could be anything - but I don't suppose it matters, and the other tests cover all the bases already.

Comment (diox) [reply src/olympia/addons/tests/test_update.py]: That's a good point, I was lazy but I'll add basic end-to-end covering the rest.

**Comment by kmaglione** [comment]
\o/

**Pull request merged**